scalac-scoverage-runtime/shared/src/main/scala/scoverage/Invoker.scala

```diff
@@ -59,6 +72,43 @@ object Invoker {
     }
   }
 
+  /**
+   * Invokes above method after adding additional features to make it work with
+   * https://github.com/scoverage/scalac-scoverage-plugin/issues/265. Typically, we
+   * add coverage instrument files [scoverage.coverage] along with the measurement files at runtime
+   * rather than compile time. This is needed as the runtime process may be running
+   * on a different host than the compile time one.
+   * However, since the instrument files are on the classpath, we can retrieve the instruments from different hosts.
+   *
+   * @param id the id of the statement that was invoked
+   * @param targetId targetId helps in grabbing the "correct" instrument file from cp that corresponds
+   *                 to the measurement file written during this invocation.
+   */
+  def invokedWriteToClasspath(id: Int, targetId: String): Unit = {
+    // Adding subdir to the basedir. This is because every
+    // unique target has a instrument file [scoverage.coverage] for which corresponding
+    // measurements [scoverage.measurements] will be generated and it is possible that a
+    // test file tests multiple targets. And thus, multiple measurement files will be generated
+    // which needs to be stored SEPARATELY from each other (but along with THEIR instrument files).
+    // Thus, to segregate (instruments + measurements) for each target, we need this subdir.
+    // This was not an issue with previous version as then we explicitly
+    // specified a different [dataDir] each time we compiled a target.
```

Comment:
Is this back to being the case? The dataDir will still be required?

Reply:
Ah, the `dataDir` from plugin.scala is the `targetId` in Invoker.scala in case of `writeToClasspath`. I should change the names so as to prevent this confusion.

Reply:
If by `dataDir` you mean the variable on line 96, then yes, you are correct - `dataDir` is no longer necessary.

Reply:
That being said, I still think its a good idea to provide an option for setting the directory prefix at runtime (using the System var in this case)

Reply:
I understand your concern regarding doing additional setup here (its effect in terms of speed)